plonky2/src/iop/generator.rs

```diff
@@ -123,13 +125,21 @@ pub trait WitnessGenerator<F: RichField + Extendable<D>, const D: usize>:
     where
         Self: Sized;
 }
+dyn_clone::clone_trait_object!(<F, const D: usize> WitnessGenerator<F, D> where F: RichField + Extendable<D>);
 
 /// A wrapper around an `Box<WitnessGenerator>` which implements `PartialEq`
 /// and `Eq` based on generator IDs.
+#[derive(Clone)]
 pub struct WitnessGeneratorRef<F: RichField + Extendable<D>, const D: usize>(
     pub Box<dyn WitnessGenerator<F, D>>,
 );
```

Comment:
Implement Clone trait for WitnessGenerator, then WitnessGeneratorRef can automatically derive Clone trait